source/upgrade.txt

```diff
@@ -258,6 +258,10 @@ The v3.11 driver introduces the following breaking changes to the ABI:
   MSVC. This is an ABI breaking change if you use an alternative
   default calling convention when building projects.
 
+- If the ``ENABLE_TESTS`` variable is set to ``ON`` while building tests in CMake,
```

Comment:
I: Since this change involves configuration options, I'd move this bullet to the "Build System Breaking Changes in v3.11" section